### DISCUSSION TITLE
Any plans to add session-replay to the roadmap?

### DISCUSSION THREAD

**Discussion by Nikola-Milovic** [SigNoz/signoz#3846] [General]
Hello great team over at SigNoz, I've checked out discussions and the roadmap, but no mention of session replay, which seems to be included with the majority of the open-source competitors. I know you guys are targeting DataDog mostly as your competitor but it would still be very useful to have this feature alongside others so that I can really only use SigNoz for my observability stack

**Comment by pranay01** [comment]
hey @Nikola-Milovic Thanks for writing here. Can you share what use cases you use session replay for? Is there an advantage of having it in your observability product vs in your analytics product ( like mixpanel, etc)

As of now, this not in our near term roadmap (next 3-4 months), but would love to understand what problems would it help you solve to have it in your observability stack?

**Reply by Nikola-Milovic** [reply]
Yeah, basically, I'd usually want to see the following things, some basic metrics, logs and alerts. SigNoz practically covers most of these, especially for the backend part, but for the frontend, another very useful tool for debugging and being ahead of issues is having session replays especially on errors/ user reports. Since I can see the state of things + their journey to the affected issue. Or even just the marketing team can observe how the users use the page and how do they navigate around it.

This is more niche towards the frontend, and I can understand that it wouldn't be a priority, but nevertheless would be an amazing thing to have alongside the other observability options

**Reply by hce137** [reply]
Hi, @pranay01, and great team, just add my two cents. I believe session-replay is a critical missing part from SigNoz to cover the observability requirement on user journey analytics, it's very helpful for product team to gain insight from user behavior to improve product usability, and customer satisfaction. It's also valuable to customer support, session replay powers support team with the holographic context of user journey to better assist customer. Developers also benefit from the full context it provides when troubleshooting. Sentry as a mature frontend APM solution, had this feature since long time ago, hyperDX(now ClickStack) as a full stack o11y solution also has this feature. Please give second thought to add this feature and maybe more. Thanks

**Comment by dmavrommatis** [comment]
Missing the Session replay feature is the reason my company wouldn't use SigNoz unfortunately.
